parquet_file/src/catalog.rs

```diff
@@ -377,6 +388,36 @@ fn parse_uuid_required(s: &str) -> Result<Uuid> {
     parse_uuid(s)?.context(UuidRequired {})
 }
 
+/// Parse [`DirsAndFilename`](object_store::path::parsed::DirsAndFileName) from protobuf.
+fn parse_dirs_and_filename(proto: &Option<proto::Path>) -> Result<DirsAndFileName> {
+    match proto {
+        Some(proto) => Ok(DirsAndFileName {
+            directories: proto
+                .directories
+                .iter()
+                .map(|s| PathPart::from(&s[..]))
+                .collect(),
+            file_name: Some(PathPart::from(&proto.file_name[..])),
+        }),
+        None => Err(Error::PathRequired {}),
+    }
+}
+
+/// Store [`DirsAndFilename`](object_store::path::parsed::DirsAndFileName) as protobuf.
+fn unparse_dirs_and_filename(path: &DirsAndFileName) -> proto::Path {
+    proto::Path {
+        directories: path
+            .directories
+            .iter()
+            .map(|part| part.encoded().to_string())
+            .collect(),
+        file_name: path
+            .file_name
+            .as_ref()
+            .map_or_else(|| "".to_string(), |part| part.encoded().to_string()),
```

Comment:
Because empty string is `String`'s default, this could be:

```suggestion
            .map(|part| part.encoded().to_string())
            .unwrap_or_default(),
```

Again, I can see reasons to prefer either, so up to you :)